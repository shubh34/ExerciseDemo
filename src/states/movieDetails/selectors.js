import get from 'lodash/get';
import { createSelector } from 'reselect';

const getMovieDetails = state => get(state, 'movieDetails', '');

export const getHighLightedScore = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.scores', []).find(score => score.highlighted));

export const getMovieTitle = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.title', ''));

export const getSnapshotImageUrl = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.images.snapshot', '').replace('.jpeg', '-width1920-quality80.jpeg'));

export const getMovieDescription = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.plot', ''));

export const getScore = createSelector(getHighLightedScore, scores => get(scores, 'score', ''));

export const getVotes = createSelector(getHighLightedScore, scores => get(scores, 'formatted_amount_of_votes', ''));
