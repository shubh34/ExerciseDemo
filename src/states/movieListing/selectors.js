import get from 'lodash/get';

export const getMovieListHeader = (state, id) => get(state, `movieListing[${id}].data.name`);

export const getMovieListByCategory = (state, id) => get(state, `movieListing[${id}].data.contents.data`, []).map(movie => ({
	artwork: movie.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
	id: movie.id,
	score: movie.highlighted_score.score,
	votes: movie.highlighted_score.formatted_amount_of_votes,
}));
