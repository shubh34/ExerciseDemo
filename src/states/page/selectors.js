export const getCurrentPage = state => state.page.name;
export const getBackPage = state => state.page.backPage;
