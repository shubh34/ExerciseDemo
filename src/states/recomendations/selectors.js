import get from 'lodash/get';
import { createSelector } from 'reselect';

export const getRecommendedMovies = state => get(state, 'recomendations.data', []);

export const getFormattedRecomendedMovieDetails = createSelector(
	getRecommendedMovies,
	movies => movies.map(movieDetails => ({
		id: movieDetails.id,
		artwork: movieDetails.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
		score: movieDetails.highlighted_score.score,
		votes: movieDetails.highlighted_score.formatted_amount_of_votes
	}))
);
