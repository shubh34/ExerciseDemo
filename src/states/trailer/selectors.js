import get from 'lodash/get';

export const getMovieTrailerUrl = state => get(state, 'trailer.trailer.data.stream_infos[0].url');
