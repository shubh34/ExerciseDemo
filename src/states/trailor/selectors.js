import get from 'lodash/get';

export const getMovieTrailorUrl = state => get(state, 'trailor.trailor.data.stream_infos[0].url');
