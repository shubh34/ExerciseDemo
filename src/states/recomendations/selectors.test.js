import { getRecommendedMovies, getFormattedRecomendedMovieDetails } from './selectors';

const state = {
	recomendations: {
		data: [
			{
				provider_id: 2,
				type: 'movies',
				id: 'pokemon-detective-pikachu',
				numerical_id: 101478,
				title: 'Pokémon: Detective Pikachu',
				display_name: 'Pokémon: Detective Pikachu',
				year: 2019,
				duration: 104,
				label: '4,99 €',
				short_plot:
					'Cuando el detective de talla mundial Harry Goodman desparece en circunstancias muy extrañas, su hijo ',
				images: {
					artwork: 'https://images-0.wuaki.tv/system/artworks/101478/master/pokemon-detective-pikachu-1564139688.jpeg',
					snapshot: 'https://images-2.wuaki.tv/system/shots/150638/original/pokemon-detective-pikachu-1564139541.jpeg',
					ribbons: [
						{
							type: 'ribbons',
							id: '6',
							numerical_id: 6,
							name: 'UHD',
							color: '#ffcc00',
							text_color: '#000000',
							position: 'top-left',
							localized_name: 'UHD',
							image: 'https://images-1.wuaki.tv/system/images/6/original/uhd-1418228576.png'
						}
					]
				},
				highlighted_score: {
					score: 6.9,
					amount_of_votes: 2471,
					formatted_amount_of_votes: '2,5K'
				},
				rating: {
					average: 4,
					scale: 5
				},
				classification: {
					type: 'classifications',
					id: '1',
					numerical_id: 1,
					name: 'T',
					age: 0,
					adult: false,
					description: 'Mostrar sólo contenidos autorizados para todos los públicos',
					default: false
				},
				labels: {
					audio_qualities: [
						{
							type: 'audio_qualities',
							id: '2.0',
							numerical_id: 2,
							name: '2.0 (Stereo)',
							abbr: '2.0',
							image: 'https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png'
						}
					],
					hdr_types: [
						{
							type: 'hdr_types',
							id: 'NONE',
							numerical_id: 1,
							name: 'SDR',
							abbr: 'NONE',
							image: null
						}
					],
					purchase_types: [
						{
							type: 'purchase_types',
							id: '1',
							numerical_id: 1,
							is_recurring: false,
							name: 'Alquiler 48H',
							label: 'ALQUILER 48H',
							kind: 'rental',
							expires_after_in_seconds: 172800,
							available_time_in_seconds: 172800
						}
					]
				}
			}
		]
	}
};
describe('Recomendation Selector', () => {
	it('should return getRecommendedMovies as empty', () => {
		expect(getRecommendedMovies({ recomendations: {} })).toMatchSnapshot();
	});
	it('should return recomended movies with details ', () => {
		expect(getRecommendedMovies(state)).toMatchSnapshot();
	});
	it('should return formated movies ', () => {
		expect(getFormattedRecomendedMovieDetails(state)).toMatchSnapshot();
	});
});
