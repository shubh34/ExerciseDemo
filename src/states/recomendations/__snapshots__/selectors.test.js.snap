// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recomendation Selector should return formated movies  1`] = `
Array [
  Object {
    "artwork": "https://images-0.wuaki.tv/system/artworks/101478/master/pokemon-detective-pikachu-1564139688-width217-quality80.jpeg",
    "id": "pokemon-detective-pikachu",
    "score": 6.9,
    "votes": "2,5K",
  },
]
`;

exports[`Recomendation Selector should return getRecommendedMovies as empty 1`] = `Array []`;

exports[`Recomendation Selector should return recomended movies with details  1`] = `
Array [
  Object {
    "classification": Object {
      "adult": false,
      "age": 0,
      "default": false,
      "description": "Mostrar sólo contenidos autorizados para todos los públicos",
      "id": "1",
      "name": "T",
      "numerical_id": 1,
      "type": "classifications",
    },
    "display_name": "Pokémon: Detective Pikachu",
    "duration": 104,
    "highlighted_score": Object {
      "amount_of_votes": 2471,
      "formatted_amount_of_votes": "2,5K",
      "score": 6.9,
    },
    "id": "pokemon-detective-pikachu",
    "images": Object {
      "artwork": "https://images-0.wuaki.tv/system/artworks/101478/master/pokemon-detective-pikachu-1564139688.jpeg",
      "ribbons": Array [
        Object {
          "color": "#ffcc00",
          "id": "6",
          "image": "https://images-1.wuaki.tv/system/images/6/original/uhd-1418228576.png",
          "localized_name": "UHD",
          "name": "UHD",
          "numerical_id": 6,
          "position": "top-left",
          "text_color": "#000000",
          "type": "ribbons",
        },
      ],
      "snapshot": "https://images-2.wuaki.tv/system/shots/150638/original/pokemon-detective-pikachu-1564139541.jpeg",
    },
    "label": "4,99 €",
    "labels": Object {
      "audio_qualities": Array [
        Object {
          "abbr": "2.0",
          "id": "2.0",
          "image": "https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png",
          "name": "2.0 (Stereo)",
          "numerical_id": 2,
          "type": "audio_qualities",
        },
      ],
      "hdr_types": Array [
        Object {
          "abbr": "NONE",
          "id": "NONE",
          "image": null,
          "name": "SDR",
          "numerical_id": 1,
          "type": "hdr_types",
        },
      ],
      "purchase_types": Array [
        Object {
          "available_time_in_seconds": 172800,
          "expires_after_in_seconds": 172800,
          "id": "1",
          "is_recurring": false,
          "kind": "rental",
          "label": "ALQUILER 48H",
          "name": "Alquiler 48H",
          "numerical_id": 1,
          "type": "purchase_types",
        },
      ],
    },
    "numerical_id": 101478,
    "provider_id": 2,
    "rating": Object {
      "average": 4,
      "scale": 5,
    },
    "short_plot": "Cuando el detective de talla mundial Harry Goodman desparece en circunstancias muy extrañas, su hijo ",
    "title": "Pokémon: Detective Pikachu",
    "type": "movies",
    "year": 2019,
  },
]
`;
